src/client/QXmppVCardManager.cpp

```diff
@@ -87,9 +87,10 @@ QXmppTask<QXmppVCardManager::VCardIqResult> QXmppVCardManager::fetchVCard(const
 QXmppTask<QXmppVCardManager::Result> QXmppVCardManager::setVCard(const QXmppVCardIq &vCard)
 {
     auto vCardIq = vCard;
-    vCardIq.setTo(client()->configuration().jidBare());
+    vCardIq.setTo({});
     vCardIq.setFrom({});
     vCardIq.setType(QXmppIq::Set);
+    d->clientVCard = vCardIq;
     return client()->sendGenericIq(std::move(vCardIq));
 }
```

Comment:
Actually my intention was to deprecate the caching of the client VCard because we don't know when it changes and we can't say if it is up-to-date. The new `fetchVCard()` also doesn't set the client vcard. Though I must admit, the documentation could be better here. And the old functions should actually be marked as deprecated.